src/server/utils/logging/attachRequestSpies.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import https from 'node:https';
+import http from 'node:http';
+import url from 'node:url';
+import onFinished from 'on-finished';
+
+import attachSpy from './attachSpy';
+
+function buildUrlObject(options, defaultProtocol) {
+  // TODO: url.parse is deprecated, use new URL() instead
+  // this is not a 1:1 replacement and will require changes.
+  // Currently the parsed url is used to provide href to
+  // loggers outgoingRequestEndSpy function.
+  // https://github.com/americanexpress/one-app/blob/main/src/server/utils/logging/setup.js#L42
+  // https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost
+  const parsedPath = url.parse(options.path || '/');
+  const protocol = options.protocol || `${defaultProtocol}:`;
+  const urlObject = {
+    auth: options.auth,
+    hostname: options.hostname || options.host || 'localhost',
+    protocol,
+    port: options.port || (protocol === 'http:' ? 80 : 443),
+    hash: parsedPath.hash,
+    path: parsedPath.path,
+    pathname: parsedPath.pathname,
+    query: parsedPath.query,
+    search: parsedPath.search,
+  };
+  if (
+    (protocol === 'http:' && urlObject.port === 80)
+    || (protocol === 'https:' && urlObject.port === 443)
+  ) {
+    urlObject.port = undefined;
+  }
+  return urlObject;
+}
```

Comment:
(ref https://github.com/americanexpress/one-app/pull/1213#discussion_r1428884493), knowing that we can use `undefined` for `port`, then there's no need for the if statement + mutation, we could simplify it like: 

```suggestion
  const urlObject = {
    auth: options.auth,
    hostname: options.hostname || options.host || 'localhost',
    protocol,
    portx: options.port || (protocol === 'http:' ? 80 : 443),
    port: options.port && (
        (protocol === 'http:' && options.port !== 80) ||
        (protocol === 'https:' && options.port !== 443)
      ) ? options.port : undefined,
    hash: parsedPath.hash,
    path: parsedPath.path,
    pathname: parsedPath.pathname,
    query: parsedPath.query,
    search: parsedPath.search,
  };

  return urlObject;
}
```

The idea is that, if the `options.port` is provided then we validate and make sure it's a different port (different than `80` for `http` and different than `443` for `https`), otherwise we just set it to `undefined`.

Not, this leads me to the following question, why do we need to unset the port for those cases? can't just use `80` for `http` since that's the provided port? looks like that we need it's just `port: options.port`

Reply:
This is not quiet the same result and I think having the removal as a separate conditional is clearer. I suspect this logic is based around old Node edge cases. 
This PR is intended as a simple lift and shift. This will need a rewrite as part of the removal `url.parse`